pennylane/devices/qutrit_mixed/measure.py

```diff
@@ -118,23 +121,39 @@ def calculate_probability(
     # if a probability is very small it may round to negative, undesirable.
     # math.clip with None bounds breaks with tensorflow, using this instead:
     probs = math.where(probs < 0, 0, probs)
-
-    if mp_wires := measurementprocess.wires:
-        expanded_shape = [QUDIT_DIM] * num_state_wires
-        new_shape = [QUDIT_DIM ** len(mp_wires)]
-        if is_state_batched:
-            batch_size = probs.shape[0]
-            expanded_shape.insert(0, batch_size)
-            new_shape.insert(0, batch_size)
-        wires_to_trace = tuple(
-            x + is_state_batched for x in range(num_state_wires) if x not in mp_wires
-        )
-
-        expanded_probs = math.reshape(probs, expanded_shape)
-        summed_probs = math.sum(expanded_probs, axis=wires_to_trace)
-        return math.reshape(summed_probs, new_shape)
-
-    return probs
+    if wires == Wires([]):
+        # no need to marginalize
+        return probs
+
+    # determine which subsystems are to be summed over
+    inactive_wires = Wires.unique_wires([wire_order, wires])
+
+    # translate to wire labels used by device
+    wire_map = dict(zip(wire_order, range(len(wire_order))))
+    mapped_wires = [wire_map[w] for w in wires]
+    inactive_wires = [wire_map[w] for w in inactive_wires]
+
+    # reshape the probability so that each axis corresponds to a wire
+    num_device_wires = len(wire_order)
+    shape = [QUDIT_DIM] * num_device_wires
+    desired_axes = math.argsort(math.argsort(mapped_wires))
+    flat_shape = (-1,)
+    expected_size = QUDIT_DIM**num_device_wires
+    batch_size = math.get_batch_size(probs, (expected_size,), expected_size)
+    if batch_size is not None:
+        # prob now is reshaped to have self.num_wires+1 axes in the case of broadcasting
+        shape.insert(0, batch_size)
+        inactive_wires = [idx + 1 for idx in inactive_wires]
+        desired_axes = math.insert(desired_axes + 1, 0, 0)
+        flat_shape = (batch_size, -1)
+
+    prob = math.reshape(probs, shape)
+    # sum over all inactive wires
+    prob = math.sum(prob, axis=tuple(inactive_wires))
+    # rearrange wires if necessary
+    prob = math.transpose(prob, desired_axes)
+    # flatten and return probabilities
+    return math.reshape(prob, flat_shape)
```

Comment:
Curious why this change was added.

Reply:
It only occurs if the number of wires that have operators is less than number of measured wires and there is a gap in the operator wires ie wire 0 and 2 have operators, but 1 doesn't